File: lib/src/tracks/better_player_track.dart
import 'dart:convert';

class BetterPlayerTrack {
  int type = 0;
  String name = "";
  int index = 0;

  BetterPlayerTrack(this.type, this.name, this.index);

  static List<BetterPlayerTrack> fromJsonArray(String jsonArrayString) {
    if(jsonArrayString.length < 3) return [];
    List<BetterPlayerTrack> list = [];
    List<dynamic> data = jsonDecode(jsonArrayString);
    data.forEach((element) {
      list.add(BetterPlayerTrack.fromJson(element));
    });
    return list;
  }

  BetterPlayerTrack.fromJson(Map<String, dynamic> json) {
    type = json['type'];
    name = json['name'];
    index = json['index'];
  }

  static int TYPE_AUDIO = 1;
  static int TYPE_VIDEO = 2;
  static int TYPE_SUBTITLE = 3;
}

File: android/src/main/kotlin/com/jhomlala/better_player/BetterPlayerTrack.kt
package com.jhomlala.better_player

class BetterPlayerTrack {
    var type: Int = 0
    var name: String = ""
    var index: Int = 0
    var isSupported: Boolean = false

    constructor(type: Int, name: String, index: Int, isSupported: Boolean) {
        this.type = type
        this.name = name
        this.index = index
        this.isSupported = isSupported
    }
}